pkg/tls/marshal.go

```diff
@@ -126,3 +159,27 @@ func ParsePrivateKey(b []byte) (*rsa.PrivateKey, error) {
 
 	return nil, errors.New("failed to parse private key")
 }
+
+// ParseCertificateChain takes slice/chain of the certificates
+// and required certificate type and return certificate from the chain
+// TODO: Needs implementation
+func ParseCertificateChain(certs []*x509.Certificate, certType CertificateType) (*x509.Certificate, error) {
```

Comment:
@jim-minter in multiple places we need to parse "CertChain" and get a certificate. Any ideas about this implementation??